ValveResourceFormat/IO/FileExtract.cs

```diff
@@ -64,67 +64,14 @@ public static ContentFile Extract(Resource resource)
 
                 case ResourceType.Texture:
                     {
-                        var bitmap = ((Texture)resource.DataBlock).GenerateBitmap();
-                        using var pixels = bitmap.PeekPixels();
-                        using var png = pixels.Encode(SKPngEncoderOptions.Default);
-                        contentFile.Data = png.ToArray();
-
-                        var spriteSheetData = ((Texture)resource.DataBlock).GetSpriteSheetData();
-
-                        if (spriteSheetData is null)
+                        var textureExtract = new TextureExtract(resource.FileName, (Texture)resource.DataBlock);
+                        if (IsChildResource(resource))
                         {
-                            bitmap.Dispose();
+                            contentFile.Data = textureExtract.ToPngImage();
```

Comment:
Rip dispose. If sheets aren't doing it, does it matter?

Reply:
The problem here is that skia has unmanaged objects, so solely relying on GC is not ideal.

Reply:
Fixed! Disposing everything now.